### DISCUSSION TITLE
Case Step, Sub-Automations, & "Trigger Automation" Step

### DISCUSSION THREAD

**Discussion by R2bEEaton** [Budibase/budibase#7553] [Ideas]
### Cases and Sub-Automations

I'm sure this has been suggested before, but something that would be really cool to see is a replacement for the "Condition" Automation step called, for example, the "Case" step. I will dive into the functionality, use cases, and consequences.

It would work similarly to a switch statement in JS. You can add as many cases as you want, for example:

Case 1. `{{ success }} = false` action `Condition 1`
Case 2. `{{ result }} not in {{ list }} && {{ length result }} = 5` action `Condition 2`
Case 3. `{{ result }} not in {{ list }} && {{ length result }} != 5` action `Stop Automation`
Case 4. `{{ result }} in {{ list }}` action `Condition 3`

Perhaps this step would be constrained such that there can only be one and it must be the last step in an automation.

Then you could create or select a sub-automation to correspond with each case, as illustrated in the image below:
![image](https://user-images.githubusercontent.com/34921506/187784310-b10b8630-2108-44d7-bb4c-2bea9c830377.png)

### Call an Automation... within an Automation

Another enhancement that might be easy to add as a consequence of this change is the ability to call completely separate automations from within an automation. This has the potential to create loops, so maybe some advanced checking for loops would need to be employed. However, the benefit of adding such a feature would be to enable calling the same automation in multiple ways (app action, row change, etc.). There are other applications for that which I'm sure are obvious.

### Conclusion, sort of

All of these changes would be immensely useful for making Automations involve less hassle. I find myself often making two very similar automations and changing the Condition step so that I can do two different things based on a certain result. I've also painstakingly recreated an automation identically, with the only difference being the activation method. I'm going through both of these things right now with my app and would really like to see Automations improved in this way in the future!

![image](https://user-images.githubusercontent.com/34921506/187790356-d184f9a1-d5fd-424a-9dd1-68804ab4c9bc.png)
Also, the Automations list should be sorted alphabetically. I had to Inspect Element to get these Automations next to each other! Maybe I'll make a separate issue for that and try my hand at it.

**Comment by JayP718** [comment]
Having trouble with this as well within Button Actions. A switch/case statement that completes certain actions dependent on certain conditions. Continue if is somewhat useful but what if two separate branches are needed. (if result is 1 do this, if result is 2 do this). From my understanding it's not currently possible with the continue if action.

**Reply by R2bEEaton** [reply]
Would be very nice there as well! I feel Automations would be a better focus of time because buttons could simply trigger an automation that could then handle all the logic. Not only that, but I think branching options would be easier to add to Automations from internal and UX design perspectives.

**Reply by JayP718** [reply]
True, the issue i'm facing is that Automations can't interact with front end components. Things like navigating to a certain form step are impossible in current state.

**Reply by chadgriff** [reply]
> Having trouble with this as well within Button Actions. A switch/case statement that completes certain actions dependent on certain conditions. Continue if is somewhat useful but what if two separate branches are needed. (if result is 1 do this, if result is 2 do this). From my understanding it's not currently possible with the continue if action.

@JayP718  I don't know if this would be applicable to your use case but I came up with the following "hack" as a way to get past the current limitations of the Continue If / Stop If logic. 

First I created a query to test for a condition (in this case I want to see if an account already exists for a given email address). 
![actions_wa1](https://user-images.githubusercontent.com/15015476/196284569-fe46695c-372f-46ac-b307-3d469664d775.png)

Then I create separate branches using pages via the Navigate To action. 
![actions-wa2](https://user-images.githubusercontent.com/15015476/196284619-8c42d7ad-d0b5-45f8-af41-eef853c869f2.png)

Then I use the Continue If / Stop If action to halt any more actions (stop from executing the query to create a new record) using the same result.
![actions_wa3](https://user-images.githubusercontent.com/15015476/196284652-b39f1d63-e302-4194-b882-62932ecd59bb.png)

**Comment by melohagan** [comment]
I quite like this because it uses a nested tree structure that users will already be familiar with from the component tree in the Design section. 

Nice mock-up!

**Comment by xnohat** [comment]
Hi I have meet a case need call automation from an automation so I using automation block "call external query" to call this api to trigger another automation from an automation.

POST https://budibase.app/api/automations/au_f6b9ee08a4be488b905fce10536ae478/trigger
{"fields":{"param":"123"}}

**Comment by IchthysMaranatha** [comment]
Have added this for actions, please look here 
https://github.com/Budibase/budibase/discussions/10123
